electrum/commands.py

```diff
@@ -1475,6 +1475,8 @@ def get_parser():
     # daemon
     parser_daemon = subparsers.add_parser('daemon', help="Run Daemon")
     parser_daemon.add_argument("-d", "--detached", action="store_true", dest="detach", default=False, help="run daemon in detached mode")
+    parser_daemon.add_argument("--rpcsock", dest="rpcsock", default='auto', help="what socket type to which to bind RPC daemon", choices=['unix', 'tcp', 'auto'])
```

Comment:
```suggestion
    parser_daemon.add_argument("--rpcsock", dest="rpcsock", default=None, help="what socket type to which to bind RPC daemon", choices=['unix', 'tcp', 'auto'])
```
This is to fix the following bug:
```
$ ./run_electrum --testnet -o setconfig rpcsock unix
true
$ ./run_electrum --testnet daemon -d
starting daemon (PID 207559)
$ cat ~/.electrum/testnet/daemon
('tcp', ('127.0.0.1', 7777), 1635959223.0871708)
$ ./run_electrum --testnet -o getconfig rpcsock
unix
```

Reply:
Thanks, didn't catch that. The correct thing is indeed to apply the default in `CommandsServer`.